vassal-app/src/main/java/VASSAL/build/module/BasicLogger.java

```diff
@@ -435,28 +455,28 @@ else if (result == 2) { // Turn Preference Off
    * Write the logfile to a file. The filename will have been selected when the logfile was begun.
    */
   public void write() throws IOException {
+    write(outputFile);
+  }
+
+  /**
+   * Write the logfile to a specified file. This is used by the autosave feature.
+   */
+  private void write(File file) throws IOException {
     if (!logOutput.isEmpty()) {
+      GameModule.getGameModule().warn(Resources.getString("BasicLogger.logging_autosaving"));  //$NON-NLS-1$
       final Command log = beginningState;
       for (final Command c : logOutput) {
         log.append(new LogCommand(c, logInput, stepAction));
       }
-
-// FIXME: Extremely inefficient! Make encode write to an OutputStream
       final String logString = GameModule.getGameModule().encode(log);
 
-      try (ZipWriter zw = new ZipWriter(outputFile)) {
+      try (ZipWriter zw = new ZipWriter(file)) {
         try (OutputStream out = new ObfuscatingOutputStream(new BufferedOutputStream(zw.write(GameState.SAVEFILE_ZIP_ENTRY)))) {
           out.write(logString.getBytes(StandardCharsets.UTF_8));
         }
         metadata.save(zw);
       }
-
-      GameModule.getGameModule().getGameState().setModified(false);
-      undoAction.setEnabled(false);
-      ModuleManagerUpdateHelper.sendGameUpdate(outputFile);
     }
-
-    endLogAction.setEnabled(false);
```

Comment:
That was a mistake, I am now keeping the original write() function as it was and creating a new write(File file) function so that the autosaver can save to a different file and not trigger the lines below when saving.

 GameModule.getGameModule().getGameState().setModified(false);
 undoAction.setEnabled(false);
 ModuleManagerUpdateHelper.sendGameUpdate(outputFile);
 endLogAction.setEnabled(false);